src/middleware/validate/validateAddressMiddleware.ts

```diff
@@ -32,17 +32,28 @@ export const validateAddressMiddleware: RequestHandler = (req, _res, next) => {
 function checkAddress(address: string): [boolean, string | undefined] {
 	let decoded;
 
-	try {
-		decoded = base58Decode(address);
-	} catch (error) {
-		return [false, (error as Error).message];
-	}
+	if (address.startsWith('0x')) {
```

Comment:
What if someone submits a normal substrate address in a raw hex form? I think we should double check that `base58Decode` would error on that. But I suspect it will validate the raw address form as well as the checksum, so if that is the case we should continue supporting those formats. If this is indeed the case I suggest we first try `base58Decode` and if that fails _then_ check for `0x` and the length of 20

Reply:
I updated the logic, can you check ?